### PR TITLE
Misc. GPU vertex loading fixes

### PR DESCRIPTION
**Pull request by yuriks**
Two small GPU fixes. See individual commit messages for more info.

Notably, one of them fixes the UI in Fire Emblem, which is now playable:

fire_emblem

Test homebrew at:
 - https://github.com/yuriks/3DS_gpu_tests/tree/color-clamping
 - https://github.com/yuriks/3DS_gpu_tests/tree/vertex-loading

### CODE REVIEW

**Comment by Subv** [comment]
LGTM

**Comment by yuriks** [comment]
Added test links to PR post.

File: src/video_core/command_processor.cpp [hunk 1]
```diff
@@ -284,6 +280,12 @@ static inline void WritePicaReg(u32 id, u32 value, u32 mask) {
                 // Send to vertex shader
                 VertexShader::OutputVertex output = VertexShader::RunShader(input, attribute_config.GetNumTotalAttributes());
 
+                // The hardware saturates vertex colors like this, *before* doing interpolation
```
Comment (neobrain) [inline src/video_core/command_processor.cpp]: Should explicitly mention that the sign is dropped.

**Comment by neobrain** [comment]
Please wait for my LGTM on this one.

**Comment by neobrain** [comment]
Could you add some comments to the hardware tests explaining what you're doing, and why? It's very hard to follow what the point of everything is currently.

It's already pretty worrying how the current trend is towards having all tests as single applications with the same boilerplate copied over and over again in no standardized format. A year from now, all those tests will be impossible to verify due to ctrulib changes and the whatnot. At least make it so that people trying to come up with a better solution don't need to spend an hour for each test case to understand what's going on.

**Comment by bunnei** [comment]
Agreed with @neobrain on the tests.... At least in that we should start thinking about collecting our GPU tests somewhere so they don't get lost in time.

Changes LGTM.

File: src/video_core/command_processor.cpp [hunk 2]
```diff
@@ -284,6 +280,12 @@ static inline void WritePicaReg(u32 id, u32 value, u32 mask) {
...
+                for (int i = 0; i < 4; ++i) {
+                    output.color[i] = float24::FromFloat32(
+                            std::fmin(std::fabs(output.color[i].ToFloat32()), 1.0f));
+                }
```
Comment (yuriks) [inline src/video_core/command_processor.cpp]: This should probably be moved to inside `RunShader` instead.

**Comment by yuriks** [comment]
Color clamping test updated with cleanups and documentation.

**Comment by neobrain** [comment]
@yuriks as for the vertex-loading test, could you check what happens if you add another render pass (before the existing two passes) where you actually render the triangles using the default attributes for the color? A theory of mine is that the default attributes might only actually be "loaded" when being used, rather than when being set through the command list.

**Comment by yuriks** [comment]
"existing two passes"? There is only one draw call per frame, I have no idea what you're talking about. Your comment makes no sense to me, the fallback attribute *is* enabled for rendering. It's still not used.

**Comment by bunnei** [comment]
Straight from the mouth of The Brain<sup>TM</sup>:
<pre>[10:51] <@neobrain> bunnei_: ogogo merge</pre>

**Comment by Subv** [comment]
fingers*
